lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1410,5 +1410,15 @@ def test_fail_cell_measure_dims(self):
             cm_dims = self.cube.cell_measure_dims(a_cell_measure)
 
 
+class Test_transpose(tests.IrisTest):
+    def setUp(self):
+        self.cube = stock.lat_lon_cube()
+
+    def test_inplace_data_transpose(self):
+        target_id = id(self.cube.lazy_data())
+        self.cube.transpose((0, 1))
+        self.assertEqual(id(self.cube.lazy_data().array), target_id)
```

Comment:
By using `.array` you're relying on a particular implementation of the transpose algorithm. I'd suggest checking the cube is still lazy with `has_lazy_data` and then resolving the numbers and checking they actually match.

Also, I'd do an actual transpose rather than deliberately doing a "null" transpose.

Putting those together:
```python
data = np.arange(12).reshape(3, 4)
cube = Cube(biggus.NumpyArrayAdapter(data))
cube.transpose()
self.assertTrue(cube.has_lazy_data())
self.assertArrayEqual(data.T, cube.data)
```